pkg/ottl/ottlfuncs/func_is_list.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type IsListArguments[K any] struct {
+	Target ottl.Getter[K]
+}
+
+func NewIsListFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("IsList", &IsListArguments[K]{}, createIsListFunction[K])
+}
+
+func createIsListFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*IsListArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("IsListFactory args must be of type *IsListArguments[K]")
+	}
+
+	return isList(args.Target), nil
+}
+
+func isList[K any](target ottl.Getter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (interface{}, error) {
```

Comment:
```suggestion
	return func(ctx context.Context, tCtx K) (any, error) {
```
I _think_ we standardized on `any` in the repo, but can't recall for sure.